8.15.1 Release Notes (#5748)
elastic/security-docs: docs/release-notes/8.15.asciidoc

[[release-notes-header-8.15.0]]
== 8.15

[discrete]
[[release-notes-8.15.1]]
=== 8.15.1

[discrete]
[[known-issue-8.15.1]]
==== Known issues

// tag::known-issue-189676[]
[discrete]
.Tags appear in Elastic AI Assistant's responses
[%collapsible]
====
*Details* +
On August 1, 2024, it was discovered that Elastic AI Assistant's responses when using Bedrock Sonnet 3.5 may include `<antThinking>` tags, for example `<search_quality_reflection>` ({kibana-pull}189676[#189676]).

====
// end::known-issue-189676[]

[discrete]
[[features-8.15.1]]
==== New features

> * Windows Image load events now include process protection status, making it easier to detect activities by both legitimate and malicious PPL activity.

Reviewer: ```suggestion
* Windows Image load events now include process protection status, making it easier to detect both legitimate and malicious PPL activity.
```

* Allows Jamf data to be examined in the visual event analyzer ({kibana-pull}190965[#190965]).

[discrete]
[[enhancements-8.15.1]]
==== Enhancements

> * Improves {elastic-defend} by reducing Malware Protection I/O and CPU usage when recently-written are subsequently executed. This change is for Windows endpoints only.

Author: Thanks, both! @gabriellandau I'm making one a minor edit to your suggested revision below to remove the dash after "recently" (that was my fault). Everything else looks good, thanks!

```suggestion
* Improves {elastic-defend} by reducing Malware Protection I/O disk and CPU usage when recently written files are subsequently executed. This update is for Windows endpoints only.
```

> * Improves {elastic-defend} by reducing Malware Protection I/O and CPU usage when recently-written are subsequently executed. This change is for Windows endpoints only.

Reviewer: We need to put `disk` before `I/O`.

> * Improves {elastic-defend} by reducing Malware Protection I/O and CPU usage when recently-written are subsequently executed. This change is for Windows endpoints only.

Author: Oops, I accidentally pasted in the wrong order. Thank you - fixed!

> * Improves {elastic-defend} by reducing Malware Protection I/O disk and CPU usage when recently written files are subsequently executed. This update is for Windows endpoints only.

Reviewer: ```suggestion
* Improves {elastic-defend} by reducing Malware Protection disk I/O and CPU usage when recently written files are subsequently executed. This update is for Windows endpoints only.
```

* Improves error handling for the Tines connector, and provides the option to use the webhook URL to connect to the Tines API ({kibana-pull}191263[#191263]).

[discrete]
[[bug-fixes-8.15.1]]
==== Bug fixes

> * Provides a slight CPU usage reduction for Windows process events, especially when the same executable is repeatedly launched, such as during compilation workloads.

Reviewer: ```suggestion
* Improved CPU usage for Windows process events when the same executable is repeatedly launched, such as during compilation workloads.
```

* Fixes an {elastic-defend} bug that affected CPU usage for Windows process events where the same executable is repeatedly launched, for example, during compilation workloads. With this fix, CPU usage is improved.
* Fixes an {elastic-defend} bug that sometimes caused malware scan response actions to crash when they attempted to scan an inaccessible directory. 
* Fixes an {elastic-defend} bug that sometimes caused {elastic-endpoint} to report an incorrect version if it used an independent {agent} release.
* Fixes an {elastic-defend} bug where the `process.thread.Ext.call_stack_final_user_module.protection_provenance_path` field might be populated with a non-path value. This fix is for Windows endpoints only.

> * Fixes an {elastic-defend} bug that can lead to Endpoint reporting `STATUS_ACCESS_DENIED` when attempting to open files for `GENERIC_READ`. Endpoint almost always recovered from this issue, but with this fix, it succeeds on the first try. This fix is for Windows endpoints only.

Reviewer 2: ```suggestion
* Fixes an {elastic-defend} bug that can lead to {elastic-endpoint} reporting `STATUS_ACCESS_DENIED` when attempting to open files for `GENERIC_READ`. {elastic-endpoint} almost always recovered from this issue, but with this fix, it succeeds on the first try. This fix is for Windows endpoints only.
```

* Fixes an {elastic-defend} regression that was introduced in 8.13.1, where security events did not populate the `user.name` field. This fix is for Windows endpoints only.

> * Fixes an {elastic-defend} bug where Endpoint sometimes missed file and network events on newer kernels that support ebpf. This only occurs if Endpoint fails to enable ebpf probes, falling back to kprobes. This fix is for Linux endpoints only.

Author: We might need to define this acronym. @gabriellandau can you verify that the definition is correct?

```suggestion
* Fixes an {elastic-defend} bug where Endpoint sometimes missed file and network events on newer kernels that support Extended Berkeley Packet Filter (eBPF). This only occurs if Endpoint fails to enable eBPF probes, and falls back to Kprobes. This fix is for Linux endpoints only.
```

Reviewer: I'm okay either way.  I think users technical enough to dig through our logs to identify the issue would know what ebpf is.  That being said, defining it here is harmless can save curious users the double click -> right click -> "Search Google"

Author: I'll hold off on defining the acronym for now and will only commit the tense-related edits that I added above. Thanks, @gabriellandau

> * Fixes a bug that caused errors to occur if you used Azure OpenAI connector for streaming ({kibana-pull}191552[#191552]).

Reviewer 2: ```suggestion
* Fixes a bug that caused errors if you used Azure OpenAI connector for streaming ({kibana-pull}191552[#191552]).
```

> * Fixes a bug the prevented duplicated prebuilt rules from inheriting **Required fields** and **Related integrations** field values ({kibana-pull}191065[#191065]).

Reviewer 2: ```suggestion
* Fixes a bug that prevented duplicated prebuilt rules from inheriting **Required fields** and **Related integrations** field values ({kibana-pull}191065[#191065]).
```

* Turns off the option to assign users to an alert if no assignees exist ({kibana-pull}190937[#190937]).
* Fixes a bug that prevented Timeline template settings from being applied to new Timelines that were generated by a rule ({kibana-pull}190511[#190511]).

> * Fixes an a bug that hid the option to select a connector for Elastic AI Assistant ({kibana-pull}189944[#189944]).

Reviewer 2: ```suggestion
* Fixes a bug that hid the option to select a connector for Elastic AI Assistant ({kibana-pull}189944[#189944]).
```
